crates/bvs-vault-router/src/contract.rs

```diff
@@ -326,6 +332,52 @@ mod execute {
                     .add_attribute("reason", data.metadata.reason),
             ))
     }
+
+    /// Cancel a resolved slashing request that an operator has already resolved the issue.
+    pub fn cancel_slashing(
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        slashing_request_id: SlashingRequestId,
+    ) -> Result<Response, ContractError> {
+        // service is the sender
+        let service = info.sender;
+
+        let slashing_request = state::get_slashing_request_by_slashing_request_id(
+            deps.storage,
+            slashing_request_id.clone(),
+        )?
+        .ok_or(ContractError::InvalidSlashingRequest {
+            msg: "No slashing request found by slashing request id".to_string(),
+        })?;
+
+        match slashing_request.status {
+            status
+                if status == SlashingRequestStatus::Finalized as u8
+                    || status == SlashingRequestStatus::Locked as u8 =>
+            {
+                return Err(ContractError::InvalidSlashingRequest {
+                    msg: "Cannot cancel a finalized or locked slashing request".to_string(),
```

Comment:
May be better to describe as 
```suggestion
                    msg: "Slash resolution window has passed, cancellation not allowed".to_string(),
```